Validate product on SKU id

thorsten-stripe opened the pull request:
The SKU id is generally the unique item identifier. We should use it to to find the correct match instead of the name.

thorsten-stripe commented:
we should also update the docs to say that a product needs to have a sku param

thorsten-stripe commented:
And a description param which we hand through to Checkout also. I can do these tmr

ChrisBrownie55 commented:
That's good, we should also add tests to ensure that we're getting this right. For example, we could write a test for two products with the same name but different SKUs and different prices.

Also @thorsten-stripe maybe we should provide a definition for SKU explaining that it needs to be unique per product that's being sold.

thorsten-stripe commented:
After working on the docs I'm unable to switch back to master :( getting:

```
error: unable to create symlink use-shopping-cart/README.md: File name too long
fatal: unable to checkout working tree
```
Have you seen this before?

Also tried cloning fresh, but getting the same error

thorsten-stripe commented:
According to https://stackoverflow.com/questions/18411200/git-unable-to-create-symlink-file-name-too-long I think this commit might be the issue: https://github.com/dayhaysoos/use-shopping-cart/commit/7f416115615707c816415344b7d40f3ba55e5514

ChrisBrownie55 commented:
@thorsten-stripe Check out #93
It seems to be a problem with Mac OSX git config for symlinks

thorsten-stripe commented:
Thanks, @ChrisBrownie55, that did the trick 👍 

I've updated the test and also added handthrough for description and image params. Now checkout picks them up also :)
![image](https://user-images.githubusercontent.com/23213994/81140824-611ab880-8f9d-11ea-929d-d87a73b460f8.png)

This should be ready to be merged and released. @dayhaysoos can you publish a new release to npm?
